File: Presentations/PEPM2020/projectile2/csharp/InputConstraints.cs
using System;
using System.IO;

public class InputConstraints {
    
    public static void input_constraints(double v_launch, double theta, double p_target) {
        StreamWriter outfile;
        outfile = new StreamWriter("log.txt", true);
        outfile.WriteLine("function input_constraints called with inputs: {");
        outfile.Write("  v_launch = ");
        outfile.Write(v_launch);
        outfile.WriteLine(", ");
        outfile.Write("  theta = ");
        outfile.Write(theta);
        outfile.WriteLine(", ");
        outfile.Write("  p_target = ");
        outfile.WriteLine(p_target);
        outfile.WriteLine("  }");
        outfile.Close();
        
        if (!(v_launch > 0)) {
            Console.Write("Warning: ");
            Console.Write("v_launch has value ");
            Console.Write(v_launch);
            Console.Write(" but suggested to be ");
            Console.Write("above ");
            Console.Write(0);
            Console.WriteLine(".");
        }
        if (!(0 < theta && theta < Math.PI / 2)) {
            Console.Write("Warning: ");
            Console.Write("theta has value ");
            Console.Write(theta);
            Console.Write(" but suggested to be ");
            Console.Write("between ");
            Console.Write(0);
            Console.Write(" and ");
            Console.Write(Math.PI / 2);
            Console.Write(" ((pi)/(2))");
            Console.WriteLine(".");
        }
        if (!(p_target > 0)) {
            Console.Write("Warning: ");
            Console.Write("p_target has value ");
            Console.Write(p_target);
            Console.Write(" but suggested to be ");
            Console.Write("above ");
            Console.Write(0);
            Console.WriteLine(".");
        }
    }
}

File: Presentations/PEPM2020/projectile2/cpp/InputConstraints.cpp
#include "InputConstraints.hpp"

#define _USE_MATH_DEFINES

#include <fstream>
#include <iostream>
#include <math.h>
#include <string>

using std::ofstream;
using std::string;

void input_constraints(double v_launch, double theta, double p_target) {
    ofstream outfile;
    outfile.open("log.txt", std::fstream::app);
    outfile << "function input_constraints called with inputs: {" << std::endl;
    outfile << "  v_launch = ";
    outfile << v_launch;
    outfile << ", " << std::endl;
    outfile << "  theta = ";
    outfile << theta;
    outfile << ", " << std::endl;
    outfile << "  p_target = ";
    outfile << p_target << std::endl;
    outfile << "  }" << std::endl;
    outfile.close();
    
    if (!(v_launch > 0)) {
        std::cout << "Warning: ";
        std::cout << "v_launch has value ";
        std::cout << v_launch;
        std::cout << " but suggested to be ";
        std::cout << "above ";
        std::cout << 0;
        std::cout << "." << std::endl;
    }
    if (!(0 < theta && theta < M_PI / 2)) {
        std::cout << "Warning: ";
        std::cout << "theta has value ";
        std::cout << theta;
        std::cout << " but suggested to be ";
        std::cout << "between ";
        std::cout << 0;
        std::cout << " and ";
        std::cout << (M_PI / 2);
        std::cout << " ((pi)/(2))";
        std::cout << "." << std::endl;
    }
    if (!(p_target > 0)) {
        std::cout << "Warning: ";
        std::cout << "p_target has value ";
        std::cout << p_target;
        std::cout << " but suggested to be ";
        std::cout << "above ";
        std::cout << 0;
        std::cout << "." << std::endl;
    }
}
